### ISSUE TITLE
Bug: `Metrics` class fails to recognize the number of samples the training dataset contains when it is of size 1

### ISSUE THREAD

**Issue by ybouilla** [fedbiomed/fedbiomed#1013]
**Bug description:**
When using the testing facility, `fedbiomed.common.metrics.Metrics` fails to recognize the number of samples of `y_pred` and `y_true` have, in the method `_configure_y_true_pred`. This raise the error  `Predictions and true values should have equal number of samples`, because the check `len(y_pred) == len(y_true)` fails, when using MNIST dataset. Indeed, `y_pred` is of size `(1, 10)`  - probability of each of the 10 digits - whereas `y_true` of size `(1,1)` so the check gives `len(y_pred)=10` and `len(y_true)=1`. Check fails albeit `y_true` and `y_pred` have correct sizes.

**Suggestions**
we can pass the number of samples as argument, or handle specifically `y_pred` and `y_true` when the testing dataset is equal to `1` or improve the batch_size inference

**Comment by srcansiz** [comment]
After checking again implementation of `_configure_y_true_pred`, I've realized that  squeezing `y_true` causes the issue. Then, I checked to see if squeezing is necessary, and finally it is not really needed. This operation is already done by  metric function provided by `scikit-learn`. For example,  `y_true = [[1]]` and `y_pred = [2]` works fine while calculating RMSE using scikit-learn. 

I tested, MNIST pytorch, MNIST sklearn, classification and regression examples with sklearn. All tests are done using `batch_size=1` and also `batch_size = >1`. Finally, it works fine after removing squeeze operation.
